File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int index =0;
        int n = nums.size();
        k %= n;
        reverse(nums.begin(),nums.end());
        reverse(nums.begin(),nums.begin()+k);
        reverse(nums.begin()+k,nums.end());
        
    }
};

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void reverse(int[] nums, int first, int last){
        while(first<last){
            int temp = nums[first];
            nums[first] = nums[last];
            nums[last] = temp;
            first++;
            last--;
        }
    }
    public void rotate(int[] nums, int k) {

        int n = nums.length-1;
        k = k % nums.length;
        reverse(nums,0,n);
        reverse(nums, 0,k-1);
        reverse(nums,k,n);

    }
}